nomenclature/core.py

```diff
@@ -16,21 +16,26 @@ class DataStructureDefinition:
     """Definition of datastructure codelists for dimensions used in the IAMC format"""
 
     def __init__(self, path):
+        """
+
+        Parameters
+        ----------
+        path : str or path-like
+            The folder with the project definitions.
+        """
         if not isinstance(path, Path):
             path = Path(path)
 
         if not path.is_dir():
             raise NotADirectoryError(f"Definitions directory not found: {path}")
 
-        self.variable = CodeList.from_directory("variable", path / "variables")
-        self.region = CodeList.from_directory(
-            "region", path / "regions", top_level_attr="hierarchy"
-        )
+        self.variable = CodeList.from_directory("variable", path / "variable")
+        self.region = CodeList.from_directory("region", path / "region")
 
-        args = [(self.region, "region"), (self.variable, "variable")]
-        for attr, name in args:
-            if not attr:
-                logger.warning(f"Attribute {name} is empty.")
+        self.dimensions = ["region", "variable"]
+        for dim in self.dimensions:
+            if not self.__getattribute__(dim):
+                logger.warning(f"Attribute '{dim}' is empty.")
```

Comment:
Does this need a warning? Are the cases where empty attributes are allowed not covered by the jsonschema?

Reply:
This is a warning not related to the attribute of a code (e.g., the variable Primary Energy), but if the entire CodeList is empty, e.g., if not a single variable is defined. This can happen if no "variable" folder is found. In this case, any validation will fail.

Might be worthwhile to make that an error.

Reply:
Ah I see, in that case, unless we want to allow a situation where no variables are defined I would vote for throwing an error.

Reply:
Done